src/core/server/elasticsearch/client/mocks.ts

```diff
@@ -139,15 +139,16 @@ export type MockedTransportRequestPromise<T> = TransportRequestPromise<T> & {
   abort: jest.MockedFunction<() => undefined>;
 };
 
-const createSuccessTransportRequestPromise = <T>(
+const createSuccessTransportRequestPromise = <T, U>(
   body: T,
-  { statusCode = 200 }: { statusCode?: number } = {}
+  { statusCode = 200 }: { statusCode?: number } = {},
+  headers?: U
 ): MockedTransportRequestPromise<ApiResponse<T>> => {
-  const response = createApiResponse({ body, statusCode });
+  const response = createApiResponse({ body, statusCode, headers });
   const promise = Promise.resolve(response);
-  (promise as MockedTransportRequestPromise<ApiResponse<T>>).abort = jest.fn();
+  (promise as MockedTransportRequestPromise<ApiResponse<T, U>>).abort = jest.fn();
 
-  return promise as MockedTransportRequestPromise<ApiResponse<T>>;
+  return promise as MockedTransportRequestPromise<ApiResponse<T, U>>;
```

Comment:
I'm not sure I'm following the type change for the new `U` type param here. The second type parameter to `ApiResponse` is for the `meta.context` field and I don't see it applied to the `headers` field at all. I think this may be unnecessary?

Reply:
@joshdover It's the only way I could actually get the headers added to the mock. If there's a better way, please let me know! I'm by no means a typescript expert.

Reply:
done.